test/ziggurat/messaging/producer_test.clj

```diff
@@ -151,15 +139,26 @@
             expected-dead-set-message (assoc message-payload :retry-count (retry-count-config))]
         (producer/retry retry-message-payload)
         (let [message-from-mq (rmq/get-msg-from-dead-queue "default")]
-          (is (= expected-dead-set-message message-from-mq))))))
+          (is (= (bytes-to-str expected-dead-set-message) (bytes-to-str message-from-mq)))))))
 
-  (testing "message with no retry count will publish to delay queue"
+  (testing "message with no retry count will publish to dead queue"
     (fix/with-queues
       {:default {:handler-fn #(constantly nil)}}
-      (let [expected-message (assoc message-payload :retry-count 4)]
-        (producer/retry message-payload)
+      (let [retry-message-payload     message-payload
+            expected-dead-set-message (assoc message-payload :retry-count (retry-count-config))]
+        (producer/retry retry-message-payload)
+        (let [message-from-mq (rmq/get-msg-from-dead-queue "default")]
+          (is (= (bytes-to-str expected-dead-set-message) (bytes-to-str message-from-mq)))))))
+
+  (testing "message with negative retry count will not be published any where"
```

Comment:
A good test!

[MINOR] Probably we should add an assertion for instant queue as well? WDYT?